File: core/src/main/java/com/core/adapter/AbstractSimpleAdapter.java
package com.core.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.ViewGroup;

import java.util.List;

/**
 * @author volkanhotur
 */

public abstract class AbstractSimpleAdapter<VH extends AbstractSimpleHolder, Data> extends RecyclerView.Adapter<AbstractSimpleHolder> {

    private List<Data> items;

    @NonNull
    @Override
    public AbstractSimpleHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull AbstractSimpleHolder abstractSimpleHolder, int i) {
        abstractSimpleHolder.bind(items.get(i));
    }

    @Override
    public int getItemCount() {
        if (items == null) {
            return 0;
        }

        return items.size();
    }
}

File: core/src/main/java/com/core/adapter/AbstractSimpleAdapter.kt
package com.core.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

/**
 * @author volkanhotur
 */
abstract class AbstractSimpleAdapter<VH : AbstractSimpleHolder<*>?, Data> : RecyclerView.Adapter<AbstractSimpleHolder<*>>() {

    private val items: List<Data>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AbstractSimpleHolder<*> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onBindViewHolder(abstractSimpleHolder: AbstractSimpleHolder<*>, i: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemCount(): Int {
        return items?.size ?: 0
    }
}